pkg/utils/utils.go

```diff
@@ -47,21 +47,21 @@ func DetectPluginWatchMode(sockDir string) bool {
 }
 
 // GetPfAddr returns SRIOV PF pci address if a device is VF given its pci address.
-// If device it not VF then this will return its own address as PF
+// If device it not VF then it will return empty string
 func GetPfAddr(pciAddr string) (string, error) {
 	pfSymLink := filepath.Join(sysBusPci, pciAddr, "physfn")
 	pciinfo, err := os.Readlink(pfSymLink)
 	if err != nil {
 		if os.IsNotExist(err) {
-			return pciAddr, nil
+			return "", nil
 		}
 		return "", fmt.Errorf("error getting PF for PCI device %s %v", pciAddr, err)
 	}
 	return filepath.Base(pciinfo), nil
 }
 
 // GetPfName returns SRIOV PF name for the given VF
-// If device is not VF then it will return its own ifname if exist else empty string
+// If device is not VF then it will return empty string
 func GetPfName(pciAddr string) (string, error) {
 	pfEswitchMode, err := GetPfEswitchMode(pciAddr)
```

Comment:
done can you have another look?

Reply:
better to use `!IsSriovVF()` as you might get a PF which is capable of SR-IOV with no VFs.

in this case, if the PF is in switchdev you might hit L#81 below and return the PF netdev instead of `""`

same might happen in `getPfAddr()` but its being prevented by lines 52-53 above (before your changes)